salt/states/archive.py

```diff
@@ -63,13 +63,26 @@ def _gen_checksum(path):
 
 
 def _checksum_file_path(path):
-    relpath = '.'.join((os.path.relpath(path, __opts__['cachedir']), 'hash'))
-    if re.match(r'..[/\\]', relpath):
-        # path is a local file
-        relpath = salt.utils.path_join(
-            'local',
-            os.path.splitdrive(path)[-1].lstrip('/\\'),
-        )
+    try:
+        relpath = '.'.join((os.path.relpath(path, __opts__['cachedir']), 'hash'))
+        if re.match(r'..[/\\]', relpath):
+            # path is a local file
+            relpath = salt.utils.path_join(
+                'local',
+                os.path.splitdrive(path)[-1].lstrip('/\\'),
+            )
+    except ValueError as exc:
+        # The path is on a different drive (Windows)
+        if str(exc).startswith('path is on'):
+            drive, path = os.path.splitdrive(path)
+            relpath = salt.utils.path_join(
+                'local',
+                drive.rstrip(':'),
+                path.lstrip('/\\'),
+            )
+        else:
+            raise
+    log.debug('Found path: %s', relpath)
```

Comment:
This is a rather generic message. Not only that, but `relpath` is just the path relative to the root of where where we are storing the checksum file. Maybe get rid of the line above, and replace the below with something like this:

```python
ret = salt.utils.path_join(__opts__['cachedir'], 'archive_hash', relpath)
log.debug('Using checksum file %s for cached archive file %s', ret, path)
return ret
```